### ISSUE TITLE
Allow compat specifiers in Pkg.add

### ISSUE THREAD

**Issue by fredrikekre** [JuliaLang/Pkg.jl#3922]
I think it would be neat to be able to do e.g.
```julia
Pkg.add(name = "Example", version = "^1.2")
```

The default meaning of `version` have `=`-semantics though which is different from the default `^`-semantics of the `[compat]` section, but perhaps that is OK.

**Comment by KristofferC** [comment]
> The default meaning of version have =-semantics though

Not really, it has the registry compat spec semantics so `version = "1.2"` allows `1.2.x`.

**Comment by KristofferC** [comment]
Dup of https://github.com/JuliaLang/Pkg.jl/issues/2524
